mermithid/processors/Sensitivity/SensitivityParameterScanProcessor.py

```diff
@@ -183,7 +183,6 @@ def InternalRun(self):
             opt = np.argmin(limit)
             rho_opt = self.rhos[opt]
             self.optimum_rhos.append(rho_opt)
-            self.sens_main.Experiment.number_density = rho_opt
```

Comment:
I was referring to this line

Reply:
Setting the experiment back to rho_opt is now done right before the prints further down to make sure they are printing the optimum density values. Between the removed line and the prints are the optional key parameter plots that also scan over density and would overwrite a setting in this line.

Reply:
We have a general issue with the sensitivity classes. The functions that calculate systematics save several parameters as class members. We did that so that we can use them elsewhere too. However that causes 2 problems:

1. The density scans end at the maximum tested density. The sensitivity class instance now has all values saved for the highest-density experiment instead of for the optimum one. Therefore, after each scan (of any parameter), the calculation has to be done once more at the optimum value. Otherwise the key parameter prints for example would print the wrong experiment. Because some values are stored as member variables it is not enough to re-set the optimum density. Instead the sensitivity calculation has to be executed for the optimum density again.
2. Some functions use values saved by other functions. So they have to be executed in the right order. Calling CL90 automatically does that. But calling other functions individually is dangerous.

Reply:
Agreed about addressing this separately. Thanks for that summary of the issue; it helped me to better understand what is going on.